search/views.py

```diff
@@ -51,7 +51,7 @@ def do_search(request, course_id=None):
     }
     status_code = 500
 
-    search_terms = request.POST["search_string"]
+    search_terms = request.POST.get("search_string", None)
```

Comment:
What is the expected behavior when search_string/search_terms is None?  I see the operation is wrapped in a try/except, but does that mean we're making a useless call out to the Elasticsearch service?  Perhaps the perform_search() function short-circuits this internally, but I wouldn't necessarily depend on that being the case?